dbms/src/Functions/formatDateTime.cpp

```diff
@@ -282,86 +296,105 @@ class FunctionFormatDateTime : public IFunction
+        if (!times)
+            return false;
+
+        const ColumnConst * pattern_column = checkAndGetColumnConst<ColumnString>(block.getByPosition(arguments[1]).column.get());
+        if (!pattern_column)
+            throw Exception("Illegal column " + block.getByPosition(arguments[1]).column->getName()
+                            + " of second ('format') argument of function " + getName()
+                            + ". Must be constant string.",
+                            ErrorCodes::ILLEGAL_COLUMN);
 
-            if (!pattern_column)
-                throw Exception("Illegal column " + block.getByPosition(arguments[1]).column->getName()
-                                + " of second ('format') argument of function " + getName()
-                                + ". Must be constant string.",
-                                ErrorCodes::ILLEGAL_COLUMN);
+        String pattern = pattern_column->getValue<String>();
 
-            String pattern = pattern_column->getValue<String>();
+        using T = typename ActionaValueTypeMap<DataType>::ActionValueType;
+        std::vector<Action<T>> instructions;
+        String pattern_to_fill = parsePattern(pattern, instructions);
+        size_t result_size = pattern_to_fill.size();
 
-            std::vector<Action<T>> instructions;
-            String pattern_to_fill = parsePattern(pattern, instructions);
-            size_t result_size = pattern_to_fill.size();
+        const DateLUTImpl * time_zone_tmp = nullptr;
+        if (arguments.size() == 3)
+            time_zone_tmp = &extractTimeZoneFromFunctionArguments(block, arguments, 2, 0);
+        else
+            time_zone_tmp = &DateLUT::instance();
 
-            const DateLUTImpl * time_zone_tmp = nullptr;
-            if (arguments.size() == 3)
-                time_zone_tmp = &extractTimeZoneFromFunctionArguments(block, arguments, 2, 0);
-            else
-                time_zone_tmp = &DateLUT::instance();
+        const DateLUTImpl & time_zone = *time_zone_tmp;
 
-            const DateLUTImpl & time_zone = *time_zone_tmp;
+        const auto & vec = times->getData();
 
-            const typename ColumnVector<T>::Container & vec = times->getData();
+        UInt32 scale [[maybe_unused]] = 0;
+        if constexpr (std::is_same_v<DataType, DataTypeDateTime64>)
+        {
+            scale = vec.getScale();
+        }
+
+        auto col_res = ColumnString::create();
+        auto & dst_data = col_res->getChars();
+        auto & dst_offsets = col_res->getOffsets();
+        dst_data.resize(vec.size() * (result_size + 1));
+        dst_offsets.resize(vec.size());
 
-            auto col_res = ColumnString::create();
-            auto & dst_data = col_res->getChars();
-            auto & dst_offsets = col_res->getOffsets();
-            dst_data.resize(vec.size() * (result_size + 1));
-            dst_offsets.resize(vec.size());
+        /// Fill result with literals.
```

Comment:
Do we better extract int into own fillWithLiterals() function?

Reply:
well, we can, but in this particular piece of code, I've just moved that outside of 'if' statement body, effectively dedenting this block, hence the difference. I would really like to refrain from refactoring of this piece to much.